### PR TITLE
fix: copy handler migrated to latest api

### PR DESCRIPTION
**Pull request by alihassan143**
#189

### CODE REVIEW

**Comment by alihassan143** [comment]
@LucasXu0

File: lib/src/service/internal_key_event_handlers/copy_paste_handler.dart [hunk 1]
```diff
@@ -26,31 +26,38 @@ Selection _computeSelectionAfterPasteMultipleNodes(
...
-      AppFlowyClipboard.setData(
-        text: textString,
-        html: htmlString,
-      );
+
+    final textNode = nodeAtPath;
+    final htmlString = nodesToHTML([textNode]);
+    String textString = "";
+    final Delta? delta = textNode.delta;
+    final childrens = textNode.children;
```
Comment (LucasXu0) [inline lib/src/service/internal_key_event_handlers/copy_paste_handler.dart:40]: no `s`. `children`

File: lib/src/service/internal_key_event_handlers/copy_paste_handler.dart [hunk 2]
```diff
@@ -26,31 +26,38 @@ Selection _computeSelectionAfterPasteMultipleNodes(
...
+    final deltaStrin = delta != null ? delta.toPlainText() : "";
```
Comment (LucasXu0) [inline lib/src/service/internal_key_event_handlers/copy_paste_handler.dart:41]: `deltaStrin` -> `plainText`

File: lib/src/service/internal_key_event_handlers/copy_paste_handler.dart [hunk 3]
```diff
@@ -26,31 +26,38 @@ Selection _computeSelectionAfterPasteMultipleNodes(
...
+    if (childrens == null) {
+      textString = deltaStrin;
     } else {
-      Log.keyboard.debug('unimplemented: copy non-text');
+      final String chilrenString = childrens.fold('', (previousValue, node) {
```
Comment (LucasXu0) [inline lib/src/service/internal_key_event_handlers/copy_paste_handler.dart:45]: `childrenString`

**Comment by LucasXu0** [comment]
Screenshot 2023-06-15 at 17 58 53

Hey, @alihassan143. The code looks good to me. However, the codecov complains. I will merge it until all the CI pass.

**Comment by LucasXu0** [comment]
```
void _handleCopy(EditorState editorState) async {}
```

You new added test doesn't cover this function.

File: test/service/internal_key_event_handlers/copy_paste_handler_test.dart [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+      // For example:
+      expect(copiedText.text, null);
+      expect(copiedText.html, null);
+    });
+  });
+
+  // Add more test cases as needed to cover different scenarios and edge cases
+}
+
+Future<void> _handleCopy(EditorState editorState) async {
```
Comment (LucasXu0) [inline test/service/internal_key_event_handlers/copy_paste_handler_test.dart:110]: You shouldn't copy this function here for testing. Instead, we should test the built-in one rather than the copied version.

Comment (LucasXu0) [reply test/service/internal_key_event_handlers/copy_paste_handler_test.dart:110]: My suggestion is following the other tests.

1. create a document.
2. simulate the copy command.
3. check the result.

**Comment by alihassan143** [comment]
@LucasXu0  kindly review the pr

File: example/assets/example.json [hunk 5]
```diff
@@ -91,7 +91,7 @@
       {
         "type": "todo_list",
         "data": {
-          "checked": false,
```
Comment (LucasXu0) [inline example/assets/example.json:94]: please don't modify the example.json

**Comment by LucasXu0** [comment]
LGTM. Please don't modify the example.json.

**Comment by alihassan143** [comment]
@LucasXu0 example.json changes reverted

**Comment by alihassan143** [comment]
@LucasXu0  any changes required ?

**Comment by alihassan143** [comment]
@LucasXu0
